keps/sig-network/20201010-kube-proxy-v2/README.md

```diff
@@ -0,0 +1,753 @@
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+At the beginning, `kube-proxy` was designed to handle the translation of Service objects to OS-level resources.
+Implementations have been userland, then iptables, and now ipvs. With the growth of the Kubernetes project, more
+implementations came to life, for instance with eBPF, and often in relation to other goals (Calico to manage
+the network overlay, Cilium to manage app-level security, metallb to provide an external LB for bare-metal clusters, etc).
+
+Along this cambrian explosion of third-party software, the Service object itself received new concepts to improve the
+abstraction, for instance to express topology. This, third-party implementation are expected to update and become more
+complex over time, even if their core doesn't change (ie, the eBPF translation layer is not affected).
+
+This KEP is born from the conviction that more decoupling of the Service object and the actual implementations is required,
+by introducing an intermediate, node-level abstaction provider. This abstraction is expected to be the result of applying
+Kubernetes' `Service` semantics and business logic to a simpler, more stable API.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- provide a node-level abstraction of the cluster-wide `Service` semantics through an API
+- allow easier, more stable proxy implementations that don't need updates when `Service` business logic changes
+- provide a client library with minimal dependencies
+- include equivalent implementations of in-project ones (userland, iptables and ipvs)
+- (optional) help proxy implementations using the same subsystem (ie iptables) to cooperate more easily
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- provide equivalent implementations of third-party ones
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Rewrite the kube-proxy to be a "localhost" gRPC API provider that will be accessible via classic TCP (`127.0.0.1:12345`)
```

Comment:
"classic TCP"? Is there a "New TCP"? :slightly_smiling_face:

Reply:
is rephrasing to `, as assual, via TCP` better?